src/core/meshcop/dataset.cpp

```diff
@@ -251,6 +244,8 @@ otError Dataset::Set(const Dataset &aDataset)
         Remove(Tlv::kDelayTimer);
     }
 
+    mUpdateTime = aDataset.GetUpdateTime();
```

Comment:
`mUpdateTime` indicates the time at which the dataset was received or retrieved from flash.  The value is used to track how much time has elapsed and is used with the Delay Timer TLV to know when the Delay Timer has expired.